jdaviz/configs/cubeviz/plugins/spectral_extraction/tests/test_spectral_extraction.py

```diff
@@ -231,3 +231,41 @@ def test_aperture_markers(cubeviz_helper, spectrum1d_cube):
         extract_plg._obj.vue_adopt_slice_as_reference()
         extract_plg._obj.vue_goto_reference_wavelength()
         assert slice_plg.slice == 1
+
+
+@pytest.mark.parametrize(
+    ('aperture_method', 'expected_flux_1000', 'expected_flux_2400', 'expected_flux_mean'),
+    [('Exact', [16.51429064, 16.52000853, 16.52572818, 16.53145005, 16.53717344, 16.54289928,
+                16.54862712, 16.55435647, 16.56008781, 16.56582186],
+      [26.812409, 26.821692, 26.830979, 26.840268, 26.849561, 26.858857,
+       26.868156, 26.877459, 26.886765, 26.896074],
+      [0.99999993, 1.00000014, 1.00000011, 0.99999987, 0.99999995, 0.99999995,
+       1.00000007, 1.00000005, 0.99999992, 0.99999996]),
+     ('Subpixel', [16.84000006] * 10, [26.92] * 10, [0.99999988] * 10),
+     ('Center', [21] * 10, [25] * 10, [1] * 10)]
+)
+def test_cone_aperture_with_different_methods(cubeviz_helper, spectrum1d_cube_largest,
```

Comment:
This is a good start. I could still improve it with an off-center target if you want though, but I am waiting for the outstanding discussions to resolve first.